darshan-util/pydarshan/darshan/experimental/plots/data_access_by_filesystem.py

```diff
@@ -0,0 +1,582 @@
+
+        if file_wr_series[filesystem] == 0:
+            ax_filesystem_counts.text(0, 0.25, ' 0 files written', transform=ax_filesystem_counts.transAxes)
+        else:
+            ax_filesystem_counts.text(0, 0.25, f' # files written ({files_written:.2E})', transform=ax_filesystem_counts.transAxes)
+
+        ax_filesystem_bytes.barh(0, bytes_written, color='red', alpha=0.3)
+        ax_filesystem_bytes.barh(1, bytes_read, color='blue', alpha=0.3)
+
+        for axis in fig.axes:
+            # hide the right side plot frame (spine) so that
+            # the value labels don't overlap with the plot frame
+            axis.spines['right'].set_visible(False)
+            axis.set_xticklabels([])
+            axis.set_yticklabels([])
+            axis.xaxis.set_ticks_position('none')
+            axis.yaxis.set_ticks_position('none')
+            axis.minorticks_off()
+
+
+def plot_with_log_file(log_file_path: str, plot_filename: str, verbose: bool = False):
+    """
+    Plot the data access by category given a darshan log
+    file path.
+
+    Parameters
+    ----------
+
+    log_file_path: path to the darshan log file
+
+    plot_filename: name of the plot file produced
+
+    verbose: if ``True``, provide extra debug information
+
+    Returns
+    -------
+
+    fig: matplotlib figure object
+    """
+    fig = plt.figure()
+    log_file = os.path.basename(log_file_path)
+    fig.suptitle(f"Data Access by Category for log file: '{log_file}'")
+    report = darshan.DarshanReport(log_file_path, read_all=True)
+    file_id_dict = report.data["name_records"]
+    filesystem_roots = identify_filesystems(file_id_dict=file_id_dict,
+                                            verbose=verbose)
+    file_rd_series, file_wr_series = unique_fs_rw_counter(report=report,
+                                                          filesystem_roots=filesystem_roots,
+                                                          file_id_dict=file_id_dict,
+                                                          processing_func=process_unique_files,
+                                                          mod='POSIX',
+                                                          verbose=verbose)
+    bytes_rd_series, bytes_wr_series = unique_fs_rw_counter(report=report,
+                                                            filesystem_roots=filesystem_roots,
+                                                            file_id_dict=file_id_dict,
+                                                            processing_func=process_byte_counts,
+                                                            mod='POSIX', verbose=verbose)
+    plot_data(fig,
+              file_rd_series,
+              file_wr_series,
+              bytes_rd_series,
+              bytes_wr_series,
+              filesystem_roots)
+
+    fig.set_size_inches(12, 4)
+    figname = f'{plot_filename}_data_access_by_category.png'
```

Comment:
This hasn't been brought up during review, but this probably isn't a good design because the `plot_filename` isn't really respected. For example:

```python
import os
from darshan.experimental.plots import data_access_by_filesystem

log_file_path = os.path.abspath('./tests/input/noposixopens.darshan')
actual_fig = data_access_by_filesystem.plot_with_log_file(log_file_path=log_file_path,
                                                          plot_filename='test.png')
```

Produces an annoying output filename: `test.png_data_access_by_category.png`

Reply:
I'm wondering if the default behavior should be to save the figure, if it would be better to leave it as an option (say, if `plot_filename` is input), or if we should just remove `fig.savefig()` and `plot_filename` all together. At the moment the other experimental plotting functions just return the figure object.